inspectit-oce-agent/src/main/java/rocks/inspectit/oce/agent/AgentMain.java

```diff
@@ -0,0 +1,113 @@
+package rocks.inspectit.oce.agent;
+
+import rocks.inspectit.oce.bootstrap.AgentManager;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.instrument.Instrumentation;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.security.AllPermission;
+import java.security.CodeSource;
+import java.security.PermissionCollection;
+import java.util.jar.JarFile;
+
+/**
+ * Entry point of the agent.
+ *
+ * @author Jonas Kunz
+ */
+public class AgentMain {
+
+    private static final String INSPECTIT_BOOTSTRAP_JAR_PATH = "/inspectit-oce-bootstrap.jar";
```

Comment:
I see you did not include the versions in here, if you want it's very simple.. push version to `version.info` file during build and add it to the `inspectit-oce-agent.jar`, then just load this as a resource before initializing the path:

```
private static final String INSPECTIT_BOOTSTRAP_JAR_PATH = "/inspectit-oce-bootstrap-{}.jar";

...

String version = ...
INSPECTIT_BOOTSTRAP_JAR_PATH.format(version)
```

I am not sure if this would help with anything actually but could be clearer.. you could also pass the version to the AgentManager..

Reply:
Tested it out, as expected it does not work: When attaching the newer agent the "version.info" from the already loaded agent is read, as this one comes first on the classpath..
I'll leave this open for #16 .